simulation.py

```diff
@@ -0,0 +1,104 @@
+
+    Return bound data, which is a dict of str (dataset names) -> Dataset.
+    """
+    raw_data = DataLoader(DATA_LIST_ALL).get_all()
+    workdays_all = {
+        region : raw_data['workdays'][region]
+        for region in TRADING_REGIONS if region in raw_data['workdays']}
+    bound_data = {}
+    for region, workdays in workdays_all.items():
+        bound_data[region] = copy.deepcopy(raw_data)
+        for dataset in bound_data[region].values():
+            dataset.set_workdays(workdays)
+    bound_data['workdays'] = workdays_all
+    return bound_data
+
+def extract_data(raw_data, data_list, price_dataset, region):
+    """
+    Extract data on data_list from raw_data.
+
+    Return data will be a dictionary, including three key-value pairs:
+    (1) 'workdays'
+        a DataController, the workdays data of the trading region.
+    (2) 'price'
+        a Dataset, the price dataset of the trading region.
+    (3) 'others'
+        a dict of str -> Dataset, other data that users specify in data_list.
+    """
+    return {
+        'workdays' : raw_data['workdays'][region],
+        'price_data' : raw_data[region][price_dataset],
+        'others' : {
+            name : raw_data[region][name]
+            for name in data_list if name in raw_data[region]}}
+
+def simulate(alpha_settings_path, skip_evaluation, output):
+    """
+    Act the process of simulating.
+    """
+    try:
+        settings = importlib.import_module(alpha_settings_path).AlphaSettings()
+    except FileNotFoundError as error:
+        raise FileNotFoundError("Alpha does not exist.") from error
+    except ImportError as error:
+        raise ImportError("No available AlphaSettings in %s"
+                          % alpha_settings_path) from error
+    data_all = initialize()
+    _, region = settings.target
+    data_required = extract_data(
+        data_all, settings.data_list,
+        TRADING_CONFIGS[settings.target][PRICE_DATASET], region)
+    cost_function = TRADING_CONFIGS[settings.target][COST_FUNCTION](
+        region, TRADING_INSTRUMENTS[settings.target])
+    results = Simulator(settings, data_required, cost_function).run()
+    if not skip_evaluation:
+        # results would be put into evaluator after import evaluator.
+        pass
+    if output:
+        data_path = os.path.join(OUTPUT_DIR, settings.alpha_name)
+        with open(data_path, 'wb') as file:
+            pickle.dump(results, file)
+    else:
+        print(results)
+
+def main():
+    """
+    Run the simulation process.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        'alpha_settings_path', help='Path of the alpha settings file.')
+    parser.add_argument(
+        '--s', '--skip_evaluation',
+        help='Running the simulation without evaluating results.',
+        action='store_true')
+    parser.add_argument(
+        '--o', '--output',
+        help='Dump simulation_results to file.',
```

Comment:
`Path to dump simulation results.`
Users don't need to know how we name simulation results in our code.